### PR TITLE
Clean admonitions in Gradle User Manual

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
Our User Manual uses too many admonitions and many of them are too long. Users will ignore them as a result.

This PR:
- Cleans admonitions to use preferred ASCII doc syntax.
- Removes long admonitions and turns them into open blocks.
- Fixes CSS of admonitions

### Reviewers: [**Build Link**](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/75426814:id/distributions/gradle-8.6-docs.zip!/gradle-8.6-20231201202801%2B0000/docs/userguide/about_manual.html)
Please make sure nothing was erroneously deleted.

### CODE REVIEW

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Review by alllex — COMMENTED** [review]
I skimmed it quickly and a couple of questions popped up

File: subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc [hunk 1]
```diff
@@ -181,12 +174,8 @@ All Kotlin DSL build scripts have implicit imports consisting of:
 ** `org.gradle.kotlin.dsl.plugins.dsl`
 ** `org.gradle.kotlin.dsl.precompile`
 
-[CAUTION]
-.Avoid using internal Kotlin DSL APIs
```
[the remark this answers is not in the document]
Comment (tresat) [reply subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc:185]: +1 for leaving the titles.  Declaring the title ahead of the `NOTE:` in the md seems fine to me.  

Unrelated, I think the title being formatted with just italics doesn't draw enough attention to it.  Perhaps the style could be made a little more emphatic in a future change.

File: subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc [hunk 2]
```diff
@@ -464,9 +437,9 @@ The `trusted-key` element works similarly to the <<sec:trusting-artifacts,truste
 - `file`, the name of the artifact _file_ to trust
 - `regex`, a boolean saying if the `group`, `name`, `version` and `file` attributes need to be interpreted as regular expressions (defaults to `false`)
 
-[WARNING]
-====
-You should be careful when trusting a key globally: try to limit it to the appropriate groups or artifacts:
+WARNING: You should be careful when trusting a key globally.
+
+Try to limit it to the appropriate groups or artifacts:
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc:442]: ❗ It looks like this line at least should be in the admonition.

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dependency_locking.adoc [hunk 3]
```diff
@@ -100,11 +91,8 @@ In order to generate or update lock state, you specify the `--write-locks` comma
 This will cause the creation of lock state for each resolved configuration in that build execution.
 Note that if lock state existed previously, it is overwritten.
 
-[NOTE]
-====
-Gradle will not write lock state to disk if the build fails.
+NOTE: Gradle will not write lock state to disk if the build fails.
 This prevents persisting possibly invalid state.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dependency_locking.adoc:95]: ❗ This text belongs in the admonition.

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc [hunk 4]
```diff
@@ -24,10 +24,7 @@ Alternatively, the module you request can change over time even for the same ver
 An example of this type of _changing module_ is a Maven `SNAPSHOT` module, which always points at the latest artifact published.
 In other words, a standard Maven snapshot is a module that is continually evolving, it is a "changing module".
 
-[NOTE]
-====
-Using dynamic versions and changing modules can lead to unreproducible builds. As new versions of a particular module are published, its API may become incompatible with your source code. Use this feature with caution!
-====
+NOTE: Using dynamic versions and changing modules can lead to unreproducible builds. As new versions of a particular module are published, its API may become incompatible with your source code. Use this feature with caution!
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc:27]: 🤔 As this note literally ends with caution, perhaps we should switch it to a `CAUTION` admonition?

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc [hunk 5]
```diff
@@ -36,10 +33,7 @@ Projects might adopt a more aggressive approach for consuming dependencies to mo
 For example you might want to always integrate the latest version of a dependency to consume cutting edge features at any given time.
 A _dynamic version_ allows for resolving the latest version or the latest version of a version range for a given module.
 
-[NOTE]
-====
-Using dynamic versions in a build bears the risk of potentially breaking it. As soon as a new version of the dependency is released that contains an incompatible API change your source code might stop compiling.
-====
+NOTE: Using dynamic versions in a build bears the risk of potentially breaking it. As soon as a new version of the dependency is released that contains an incompatible API change your source code might stop compiling.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/dynamic_versions.adoc:36]: 🤔 Another caution?

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc [hunk 6]
```diff
@@ -94,12 +87,9 @@ include::sample[dir="snippets/dependencyManagement/managingTransitiveDependencie
 
 While the previous section showed how you can enforce a certain version of a transitive dependency, this section covers _excludes_ as a way to remove a transitive dependency completely.
 
-[WARNING]
-====
-Similar as forcing a version of a dependency, excluding a dependency completely requires a conscious decision.
+WARNING: Similar as forcing a version of a dependency, excluding a dependency completely requires a conscious decision.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/dependency_downgrade_and_exclude.adoc:90]:
```suggestion
WARNING: Similar to forcing a version of a dependency, excluding a dependency completely requires a conscious decision.
```

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc [hunk 7]
```diff
@@ -216,21 +213,16 @@ include::sample[dir="snippets/dependencyManagement/catalogs-settings/kotlin",fil
 include::sample[dir="snippets/dependencyManagement/catalogs-settings/groovy",files="settings.gradle[tags=extra_catalog]"]
 ====
 
-[NOTE]
-====
-Each catalog will generate an extension applied to all projects for accessing its content.
+NOTE: Each catalog will generate an extension applied to all projects for accessing its content.
 As such it makes sense to reduce the chance of collisions by picking a name that reduces the potential conflicts.
```
Comment (tresat) [inline subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc:217]: ❗ This text continues the text in the admonition.

**Review by tresat — CHANGES_REQUESTED** [review]
I stopped reviewing after about a third of the way.

I completely agree there are way too many admonitions, and that many of the longer ones should be removed and incorporated into the normal text flow, or at least shortened.  

However, removing all but the first sentence of some of these notes from the admonitions itself seems to be often leading to the following pattern:


```markdown
Text about some topic X...

NOTE: By the way, also Y...

// Note ends

Therefore, Y means that...  // <-- This is more about Y and doesn't make sense outside of the admonition

Moving on to topic Z...
```

I think a rule of thumb should be the admonitions are self-contained to a large extent.  So if you remove them, does the text still make sense and continue to flow naturally?  If the answer is no, I think some of the _after_ text should be rewritten or moved into the admonition itself.

File: subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc:185]: Admonitions should to be short and sweet. They should not have a title. 
Block should have a title.
I'll make sure to update this PR accordingly.

File: subprojects/docs/src/docs/userguide/api/groovy_build_script_primer.adoc [hunk 8]
```diff
@@ -37,12 +37,8 @@ Both `version` and `configurations {}` are part of link:{groovyDslPath}/org.grad
 
 This example reflects how every Groovy build script is backed by an implicit instance of `Project`. If you see an unqualified element and you don't know where it's defined, always check the `Project` API documentation to see if that's where it's coming from.
 
-[CAUTION]
-====
-Avoid using https://groovy-lang.org/metaprogramming.html#_metaclasses[Groovy MetaClass] programming techniques in your build scripts. Gradle provides its own API for adding <<writing_build_scripts#sec:extra_properties,dynamic runtime properties>>.
-
+CAUTION: Avoid using https://groovy-lang.org/metaprogramming.html#_metaclasses[Groovy MetaClass] programming techniques in your build scripts. Gradle provides its own API for adding <<writing_build_scripts#sec:extra_properties,dynamic runtime properties>>. +
```
[the remark this answers is not in the document]
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/api/groovy_build_script_primer.adoc:40]: + is a forced new line in ascii docs.

File: subprojects/docs/src/docs/userguide/authoring-builds/writing_build_scripts.adoc [hunk 9]
```diff
@@ -20,17 +20,10 @@ that needs to be built, like a library or an application. Each build script is a
...
-====
-_Build scripts_ define `Project` objects.
-Gradle uses other scripts as well:
-
-- <<organizing_gradle_projects.adoc#sec:settings_file,Settings scripts>>
-  define link:{groovyDslPath}/org.gradle.api.initialization.Settings.html[Settings] objects.
-- <<init_scripts.adoc#init_scripts,Init scripts>>
-  define link:{groovyDslPath}/org.gradle.api.invocation.Gradle.html[Gradle] objects.
-====
+IMPORTANT: Build scripts_ define `Project` objects. +
```
[the remark this answers is not in the document]
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/authoring-builds/writing_build_scripts.adoc:23]: Thank you

File: subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc [hunk 10]
```diff
@@ -15,14 +15,11 @@
 [[resolution_rules]]
 = Customizing resolution of a dependency directly
 
-[WARNING]
-====
-This section covers mechanisms Gradle offers to directly influence the behavior of the dependency resolution engine.
+WARNING: This section covers mechanisms Gradle offers to directly influence the behavior of the dependency resolution engine.
```
[the remark this answers is not in the document]
Comment (lkasso) [reply subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc:18]: Great question.

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@bot-gradle test and merge

**Pull request merged**